mesonbuild/linkers/linkers.py

```diff
@@ -349,15 +349,25 @@ class VisualStudioLinker(VisualStudioLikeLinker, StaticLinker):
 
     """Microsoft's lib static linker."""
 
+    id = 'lib'
```

Comment:
I think you might be able to reduce your diff by adding these ids in a separate commit that could just be pulled into master.

Reply:
This feature directly requires this change.  I'm not sure what problem it gives by having it included here (perhaps you can explain your thinking)... While I am sure that the re-shuffling and unpicking of this and related changes, to make a separate commit for a separate PR, which this one would then depend on, is quite a lot of potentially error-prone jiggling around and paper shuffling for an unclear purpose and benfit.

Reply:
You're only hurting your own chances of getting this PR merged. Up to you.

It doesn't have to be a separate PR, just another commit.